subsys/debug/gdbstub/gdbstub_serial.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
```

Comment:
Maybe have backend in the file name, it is worth it looking at how tracing and other subsystems implement backends from different transports. Long term goal would be to reuse those instead of having each subsystem (tracing, loggin, debugging, ...) use there own backend implementation of the same transport...